tfjs-converter/src/executor/model_analysis.ts

```diff
@@ -146,26 +145,87 @@ export function getNodesInTopologicalOrder(
   return orderedNodes;
 }
 
-const CONTROL_FLOW_OPS = [
+/**
+ * Given the execution info, return a map from node to the disposable node list
+ * after its execution.
+ *
+ * @returns A map from node to disposable nodes after its
+ *     execution. That is, for a node `x`, `nodeLiveUntilMap[x]` indicates all
+ *     nodes which their intermediate tensors should be disposed after `x` being
+ *     executed.
+ */
+export function getNodeLiveUntilMap(orderedNodes: Node[]): Map<Node, Node[]> {
+  const nNodes = orderedNodes.length;
+  const nodeToOrder = new Map<Node, number>();
+  for (let i = 0; i < nNodes; ++i) {
+    nodeToOrder.set(orderedNodes[i], i);
+  }
```

Comment:
Tiny nit, feel free to ignore, but this may be more readable.

```suggestion
  const nodeToOrder = new Map(orderedNodes.map((node, index) => [node, index]));
```